website/blog/2024-08-14-otel-context-propagation-in-ferretdb.md

```diff
@@ -0,0 +1,160 @@
+This field doesn't influence query logic but can be used to provide valuable context.
+
+At FerretDB, we want to empower users to pass such context.
+We decided to utilize the `comment` field to do it.
+We parse the content of the `comment` field, and if it's a JSON document with a `ferretDB` key, we check for tracing data.
+If the tracing data is present, FerretDB sets the parent's context when a span for the current operation is created.
+
+An example of a comment with tracing data could look like this:
+
+```json
+{
+  "ferretDB": {
+    "traceID": "1234567890abcdef1234567890abcdef",
+    "spanID": "fedcba9876543210"
+  }
+}
+```
+
+The strings `traceID` and `spanID` are hex-encoded strings that represent the corresponding trace and span identifiers.
+
+Using this approach, it's relatively easy to visualize client's requests to FerretDB showing them as part of the original client trace.
+
+However, there are some limitations.
+Not all operations support the comment field.
+For instance, operations like `insert` or `listCollections` do not support it.
+
+Another challenge with the `comment` field is its handling in some MongoDB drivers, where it is typed as a string.
+This means that, in principle, any string can be passed.
+In our scenario, where we allow FerretDB to receive and parse
+tracing data through the `comment` field, the JSON must be correctly encoded by the driver and then decoded by FerretDB.
+
+A more robust solution would involve having a dedicated field for passing request-related context,
+and it would be ideal to establish a standard for such fields.
+For instance, it could be a BSON document with particular tracing-related fields.
+This would provide a more reliable method for passing context to the database.
+
+## Example application
+
+Since such a standard does not yet exist, let's explore [an example application](https://gist.github.com/rumyantseva/3c6ef7c7dfc3fbdea8f94a31f4a17885)
+that interacts with FerretDB and passes the tracing context using the `comment` field.
+
+For simplicity, most error handling is omitted in the example below.
+
+```go
+collection := client.Database("testdb").Collection("customers")
+
+user := bson.D{
+    {Key: "name", Value: "John Doe"},
+    {Key: "email", Value: "john.doe@example.com"},
+}
+
+insertCtx, insertSpan := tracer.Start(ctx, "InsertCustomer")
+_, _ = collection.InsertOne(insertCtx, user)
+insertSpan.End()
+
+findCtx, findSpan := tracer.Start(ctx, "FindCustomer")
+filter := bson.D{{Key: "name", Value: "John Doe"}}
+
+var result bson.D
+_ = collection.FindOne(findCtx, filter).Decode(&result)
+
+findSpan.End()
+```
+
+In this setup, we create two spans: one for inserting a document and another for finding it.
+
+If we run this application, we will see that the spans for the `InsertCustomer` and `FindCustomer` operations are created:
+
+[![Trace without propagation](/img/blog/ferretdb-otel/without-propagation.png)](/img/blog/ferretdb-otel/without-propagation.png)
```

Comment:
I think the format should be like `![Trace without propagatio](/img/blog/ferretdb-otel/without-propagation.png)`.

Clicking image currently behaves weird. Also check other image